manifests/forwarder.pp

```diff
@@ -153,10 +157,17 @@
   Package[$package_name] -> Splunkforwarder_transforms<||> ~> Service[$virtual_service]
   Package[$package_name] -> Splunkforwarder_web<||>        ~> Service[$virtual_service]
 
-  File {
-    owner => $splunk_user,
-    group => $splunk_user,
-    mode => '0644',
+  if $::osfamily == 'windows' {
```

Comment:
this should also be rewritten to the selector pattern. An example is here: https://github.com/voxpupuli/puppet-splunk/pull/114/files. In my opinion it's a bit more beautiful to write the result of the selector into a variable, and use that in the file resource declaration.

----

This should be rewritten to only declare the file resource once